js/preview/src/dev.ts

```diff
@@ -109,13 +109,47 @@ function to_posix(_path: string): string {
 	return _path.replace(/\\/g, "/");
 }
 
-function generate_imports(component_dir: string, root: string): string {
+export interface ComponentConfig {
+	plugins: any[];
+	svelte: {
+		preprocess: unknown[];
+	};
+}
+
+async function generate_imports(
+	component_dir: string,
+	root: string
+): Promise<[string, ComponentConfig]> {
 	const components = find_frontend_folders(component_dir);
 
 	const component_entries = components.flatMap((component) => {
 		return examine_module(component, root, "dev");
 	});
 
+	let component_config = {
+		plugins: [],
+		svelte: {
+			preprocess: []
+		}
+	};
```

Comment:
However for dev mode, this is one process for all components, so we can only use one of these config files.

Ideally we'd put this in the root but that isn't where we install the packages, so this won't really work (no node_modules).

Reply:
In hindsight, it might have been better to put the package.json in the root and install the node_modules there but this would be a breaking change now.

Reply:
What we could do is check for a config file in the frontend dirs, then recursivelky walk up the tree. And check each dir above. This is how most config files in the JS world work and how node_modules works.

This would give us something that works in the default case but allow people with more complex setups to add a single config file to a shared root. It would be on them to add a package.json in the root but this might be nice for more complex setups.